### ISSUE TITLE
[v4][iOS] topBar largeTitle transation too long

### ISSUE THREAD

**Issue by djschilling** [wix/react-native-navigation#5793]
When using the largeTitle and then navigating to another screen, the large title is visible all the time until the next screen is fully visible. This looks really wrong.

### Steps to Reproduce / Code Snippets / Screenshots

Link to a video recording of the problem: https://seafile.church.tools/f/3b9aa63ee1294be591b8/

---
### Environment
* React Native Navigation version: 4.0.6 and 3.7.0
* React Native version: 0.61.5
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator iPhone 11, Device iPhone X

**Comment by kuhnza** [comment]
I'm also seeing this on version 4.5.3

**Comment by kuhnza** [comment]
I found a clue to this issue in this article https://www.morningswiftui.com/blog/fix-large-title-animation-on-ios13

Turns out this is expected behaviour in the native component when transitioning from a ViewController that has `prefersLargeTitles` set to `true` to one that has it set to `false`. This from the article:

> If you are using navigationItem.largeTitleDisplayMode you basically always have to set prefersLargeTitles property to be true.
> 
> The reason is in the documentation itself, if this property is false then largeTitleDisplayMode has no effect, and it will always use the small title which will break the animation.

I was able to resolve this in my RN app by ensuring that every screen I pushed from the root one with `largeTitle: {visible: true}` set also had this set in its options. Animation works perfectly now.

It might be worth updating the docs to explain this behaviour?
